### PR TITLE
Now using custom assertions for test.

### PR DESCRIPTION
**Pull request by dipak-pawar**
Now tests are using custom assertions

### CODE REVIEW

File: src/test/java/test/integration/algeron/AddArquillianAlgeronRetrieverTest.java [hunk 1]
```diff
@@ -23,15 +20,8 @@ public void should_register_folder_retriever() throws TimeoutException {
...
-        assertThat(arquillianXml.exists()).isTrue();
-
-        final ArquillianConfig arquillianConfig = new ArquillianConfig(arquillianXml.getResourceInputStream());
-        assertThat(arquillianConfig.isExtensionRegistered("algeron-provider")).isTrue();
-        final String configuration = arquillianConfig.getContentOfNode("extension@qualifier=algeron-provider/property@name=retrieverConfiguration");
-        assertThat(configuration).isNotNull();
-        assertThat(configuration)
-            .contains("provider: folder", "contractsFolder: /tmp");
+        assertThat(project).hasArquillianConfig().withExtension("algeron-provider")
+            .withProperty("retrieverConfiguration", "provider: folder\ncontractsFolder: /tmp");
```
Comment (bartoszmajsak) [inline src/test/java/test/integration/algeron/AddArquillianAlgeronRetrieverTest.java]: I think previous way was more readable. What if we add `\n\n` which should be conceptually irrelevant? We shouldn't match strings as such, but key-values, right? Can you think about making it even more pleasant to use?

File: src/test/java/test/integration/support/assertions/ArquillianConfigAssert.java [hunk 2]
```diff
@@ -33,6 +40,16 @@ public NodeAssert(Node actual) {
             super(actual, NodeAssert.class);
         }
 
+        public NodeAssert hasConfiguration() {
+            final List<Node> nodes = actual.getChildren();
+            final Node configuration = nodes.stream()
+                .filter(node -> node.getName().equals("configuration"))
+                .map(Function.identity())
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException("Node with configuration not found."));
```
Comment (bartoszmajsak) [inline src/test/java/test/integration/support/assertions/ArquillianConfigAssert.java]: Or maybe `AssertionError` instead?

**Review by bartoszmajsak — CHANGES_REQUESTED** [review]
Overall nice improvement, thanks for spending time on it! There are just two things I would like you to consider before we merge it.

File: src/test/java/test/integration/support/assertions/ArquillianConfigAssert.java (hunk 2 above)
Comment (dipak-pawar) [reply src/test/java/test/integration/support/assertions/ArquillianConfigAssert.java]: :+1:

File: src/test/java/test/integration/support/assertions/ArquillianConfigAssert.java [hunk 3]
```diff
@@ -64,5 +83,14 @@ private String getContentOfNode(String property, List<Node> nodes) {
                 .findFirst()
                 .orElseThrow(() -> new IllegalStateException("Node with property: " + property + " not found."));
         }
+
+        public ObjectArrayAssert<String> withProperty(String key) {
+            final String contentOfNode = getContentOfNode(key, actual.getChildren());
+            Assertions.assertThat(contentOfNode).isNotEmpty();
+
+            final String[] strings = contentOfNode.trim().split("\n");
```
Comment (bartoszmajsak) [inline src/test/java/test/integration/support/assertions/ArquillianConfigAssert.java]: We create configuration file with new lines using `System.lineSeparator()`. I'm afraid splitting by `\n` would not always work.

**Review by bartoszmajsak — APPROVED** [review]

**Pull request merged**